code/game/objects/machinery/computer/camera_advanced.dm

```diff
@@ -158,6 +160,55 @@
 	eyeobj.setLoc(eyeobj.loc)
 
 
+/obj/machinery/computer/camera_advanced/proc/track(mob/living/target)
+	if(!istype(target))
+		return
+
+	INVOKE_ASYNC(src, .track_actual, target)
+
+
+
+/obj/machinery/computer/camera_advanced/proc/track_actual(mob/living/target)
+	set waitfor = FALSE
+
+	tracking_target = target
+	tracking = TRUE
+
+	if(!target || !target.can_track(current_user))
+		to_chat(current_user, "<span class='warning'>Target is not near any active cameras.</span>")
+		tracking_target = null
+		return
+
+	to_chat(current_user, "<span class='notice'>Now tracking [target.get_visible_name()] on camera.</span>")
+
+	var/cameraticks = 0
+	while(tracking_target == target)
```

Comment:
This needs to be done with standard processing from a subsystem

Reply:
You invoked the dreaded ninjanomnom reviews, which are much more demanding than mine.
May this be a lesson to never wake up this creature from its slumber, in his house at R'lyeh.

Reply:
~~I should just find a way to call the ai tracking proc~~

Reply:
Is a subsystem really necessary for something as simple as this tracking? It already uses stoplag(), it won't be used much per round.